test/local/routes/sign.js

```diff
@@ -4,52 +4,89 @@
 
 'use strict'
 
+const ROOT_DIR = '../../..'
+
 const assert = require('insist')
-var crypto = require('crypto')
-var uuid = require('uuid')
-var getRoute = require('../../routes_helpers').getRoute
-var mocks = require('../../mocks')
-var P = require('../../../lib/promise')
+const crypto = require('crypto')
+const uuid = require('uuid')
+const getRoute = require('../../routes_helpers').getRoute
+const mocks = require('../../mocks')
+const P = require(`${ROOT_DIR}/lib/promise`)
 
 describe('/certificate/sign', () => {
-  var deviceId = crypto.randomBytes(16).toString('hex')
-  var mockDevices = mocks.mockDevices({
-    deviceId: deviceId
-  })
-  const mockLog = mocks.spyLog()
-  const mockRequest = mocks.mockRequest({
-    credentials: {
+  let db, deviceId, mockDevices, mockLog, sessionToken, mockRequest
+
+  before(() => {
+    db = mocks.mockDB()
+    deviceId = crypto.randomBytes(16).toString('hex')
+    mockDevices = mocks.mockDevices({
+      deviceId: deviceId
+    })
+    mockLog = mocks.spyLog()
+    const Token = require(`${ROOT_DIR}/lib/tokens/token`)(mockLog)
+    const SessionToken = require(`${ROOT_DIR}/lib/tokens/session_token`)(mockLog, Token, {
+      tokenLifetimes: {
+        sessionTokenWithoutDevice: 2419200000
+      }
+    })
+    return SessionToken.create({
```

Comment:
Modified these tests to create a real session token, so that we can add some coverage for the call to `sessionToken.setUserAgentInfo` by asserting the arguments to `db.updateSessionToken`.